object-filter/src/main/java/org/infinispan/objectfilter/impl/RejectObjectFilter.java

```diff
@@ -23,7 +23,11 @@ public ObjectFilter withParameters(Map<String, Object> namedParameters) {
       if (namedParameters == null) {
          throw new IllegalArgumentException("namedParameters argument cannot be null");
       }
-      //todo validate params
+      for (String paramName : getParameterNames()) {
+         if (namedParameters.get(paramName) == null) {
+            throw new IllegalArgumentException("Query parameter '" + paramName + "' was not set");
```

Comment:
Here as well. It's the 3rd time I'm seeing this loop - would it be possible to extract it somehow?